File: 434. Number of Segments in a String/434. Number of Segments in a String.js
/**
 * @param {string} s
 * @return {number}
 */
var countSegments = function(s) {
    var res = 0;
    for(var i = 0; i < s.length; i++) {
        if((s[i] !== " ") && (i === 0 || s[i-1] === " ")) res++;
    }
    return res;
};

File: 434. Number of Segments in a String/434. Number of Segments in a String.rb
# @param {String} s
# @return {Integer}
def count_segments(s)
    s.strip!
    res = 0
    s.length.times do |i|
    	if (s[i] != " ") && ((i == 0) || (s[i-1] == " "))
    		res += 1
    	end
    end
    res
end
